clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -321,6 +332,9 @@ public void onHeartbeatResponseReceived(ConsumerGroupHeartbeatResponseData respo
         this.memberEpoch = response.memberEpoch();
         ConsumerGroupHeartbeatResponseData.Assignment assignment = response.assignment();
 
+        clientTelemetryReporter.ifPresent(reporter -> reporter.updateMetricsLabels(
+            Collections.singletonMap(ClientTelemetryProvider.GROUP_MEMBER_ID, memberId)));
```

Comment:
This will work, but you are updating the metrics labels on every single heartbeat response. The member ID and member epoch are always sent.

Reply:
Yeah, I needed that feedback and that's the reason I saw multiple updates happening for labels and optimized duplicate labels.
However I think handling duplicates was anyways required but let me check what should be the right place where this update should be called.

Reply:
Rechecked the code and it seems the placement of update code is right but I can optimize with the check that there occurred a change in `memberId` prior updating it. Making the change.

Reply:
@AndrewJSchofield I have added the check with comment, please see if it makes sense.